test/utils/pickers/createPickerRenderer.tsx

```diff
@@ -14,9 +15,30 @@ export function createPickerRenderer({
   locale,
   adapterName,
   instance,
+  clock: inClock,
+  clockConfig,
   ...createRendererOptions
 }: CreatePickerRendererOptions = {}) {
-  const { clock, render: clientRender } = createRenderer(createRendererOptions);
+  // TODO: Temporary until vitest is enabled
+  // If only clockConfig='2020/02/20' is provided, we just fake the Date, not the timers
+  // Most of the time we are using the clock we just want to fake the Date
+  // If timers are faked it can create inconsistencies with the tests.
+  // In some cases it also prevents us from really testing the real behavior of the component.
+  if (!inClock && clockConfig) {
+    let timer: sinon.SinonFakeTimers | null = null;
+    beforeEach(() => {
+      timer = sinon.useFakeTimers({ now: clockConfig, toFake: ['Date'] });
```

Comment:
Have you tried using the additional option added in https://github.com/mui/material-ui/pull/43729? 🤔